client/src/common/client.ts

```diff
@@ -331,7 +331,8 @@ export type LanguageClientOptions = {
 	};
 	workspaceFolder?: VWorkspaceFolder;
 	connectionOptions?: {
-		cancellationStrategy: CancellationStrategy;
```

Comment:
I don't think this is required since connectionOptions is just passed in to createProtocolHandler which doesn't require cancellationStrategy. (maxRestartCount is used in BaseLanguageClient constructor though so I guess that's why connectionOptions doesn't have type ConnectionOptions)

Reply:
It is used here: https://github.com/microsoft/vscode-languageserver-node/blob/8b56a58d0a706bc0cc8e947b42ad135faf488fe3/jsonrpc/src/common/connection.ts#L540 and travels down their via structural typing.